tests/integration/standard/test_application_info.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+
+from cassandra.application_info import ApplicationInfo
+from tests.integration import TestCluster, notscylla
+
+
+@notscylla
+class ApplicationInfoTest(unittest.TestCase):
+    def setUp(self):
+        self.cluster = TestCluster(application_info=ApplicationInfo(
+            application_name='TestApplicationInfo',
+            application_version='1.0.0-test',
+            client_id='some-client-id',
+        ))
+        self.session = self.cluster.connect()
+
+    def tearDown(self):
+        self.cluster.shutdown()
+
+    def test_application_info_endup_in_system_views_clients(self):
+        """
+        Test to ensure that ApplicationInfo user provides endup in `client_options` of `system_views.clients` table
+        """
+        found = False
+
+        for row in self.session.execute("select client_options from system_views.clients"):
+            if not row['client_options']:
+                continue
+            for marker in ("'APPLICATION_NAME': 'app_name'", "'APPLICATION_VERSION': 'app_version'", "'CLIENT_ID': 'client_id'"):
+                if marker not in row['client_options']:
+                    break
+            else:
+                found = True
+        assert found
```

Comment:
Don't you think some more tests are required here? For example, previous iteration of this PR had a bug when creating ApplicationInfo with some parameters absent. Can you think of some other test cases?

Reply:
added more tests